### PR TITLE
CPM-529: Add SetDateValue UserIntent

### PR DESCRIPTION
**Pull request by cemorin**
**Description (for Contributor and Core Developer)**
Add SetDateValue UserIntent

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php [hunk 1]
```diff
@@ -307,7 +328,7 @@ private function assertProductHasCorrectValueByAttributeCode(string $attributeCo
         Assert::assertNotNull($product);
         $value = $product->getValue($attributeCode, null, null);
         Assert::assertNotNull($value);
-        Assert::assertSame($expectedValue, $value->getData());
+        Assert::assertEquals($expectedValue, $value->getData());
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php:331]: assertSame doesn't work with object comparison

File: src/Akeneo/Pim/Enrichment/Product/back/API/Command/UserIntent/SetDateValue.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class SetDateValue implements ValueUserIntent
+{
+    public function __construct(
+        private string $attributeCode,
+        private ?string $channelCode,
+        private ?string $localeCode,
+        private \DateTime $value
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Product/back/API/Command/UserIntent/SetDateValue.php:17]:
```suggestion
        private \DateTimeInterface $value
```

Using interface is better

File: src/Akeneo/Pim/Enrichment/Product/back/API/Command/UserIntent/SetDateValue.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    ) {
+    }
+
+    public function value(): \DateTime
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Product/back/API/Command/UserIntent/SetDateValue.php:21]:
```suggestion
    public function value(): \DateTimeInterface
```

**Review by nmarniesse — APPROVED** [review]
Using interface is better. It allows to use an instance of \DateTimeImmutable that is the same than \DateTime except the time can not be updated

**Review by amigaire — APPROVED** [review]

**Pull request merged**